server/utils/util.go

```diff
@@ -0,0 +1,119 @@
+package utils
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+)
+
+var l = log.New(os.Stderr, "", 1)
+
+// Logs error to Stderr.
+func LogError(a ...interface{}) {
+	l.Println(a...)
+}
+
+// Generates Random string using crypto/rand.
+func CryptoRandomString(len int) (string, error) {
+	random := make([]byte, len)
+	_, err := rand.Reader.Read(random)
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(random), nil
+}
+
+// Enables content security policy which only allows self scripts as well as specific nonces for inline scripts.
+func EnableContentSecurityPolicy(w *http.ResponseWriter) {
+	ContentSecurityPolicyString := strings.Join([]string{
+		"script-src 'self' 'nonce-EDNnf03nceIOfn39fn3e9h3sdfa' 'nonce-afds3h9e3nf93nf0Iecn30fnNDE'",
+		"style-src 'self' 'unsafe-inline' fonts.googleapis.com ",
+		"object-src 'none'"}, ";")
+	(*w).Header().Set("Content-Security-Policy", ContentSecurityPolicyString)
+}
+
+// Gets header from request based on key.
+func GetHeader(r *http.Request, key string) string {
+	return r.Header.Get(key)
+}
+
+// Extracts a token from a cookie based on token name.
+func ExtractTokenFromCookie(tokenName string, r *http.Request) (string, error) {
+	cookie, err := r.Cookie(tokenName)
+	if err != nil {
+			return "", errors.New(tokenName + " not found")
+	}
+	return cookie.Value, nil
+}
```

Comment:
nit: this function also seems redundant

Reply:
This function is a helper that increases code simplicity/readability in other areas. I can remove it if requested?

Reply:
IMO it's not much simpler to write:
```go
token, err := ExtractTokenFromCookie(tokenName, r)
if err != nil {
  return err
}
// use token
```
than
```go
tokenCookie, err := r.Cookie(tokenName)
if err != nil {
  return err
}
// use tokenCookie.Value
```

As for readability - a reader needs to jump through a layer of indirection to understand what the code does (if they don't completely trust the function name).

Reply:
@awly I see, what you are saying makes sense. I will make the change.